src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -19,7 +19,7 @@ public ShoppingCart add(ShoppingCart shoppingCart) {
         try {
             session = HibernateUtil.getSessionFactory().openSession();
             transaction = session.beginTransaction();
-            transaction.begin();
+            shoppingCart.setUser(session.merge(shoppingCart.getUser()));
```

Comment:
I don't think you should set user in dao level. It's better to do it in service when you try to register the new shopping cart.